### PR TITLE
vtxo: classify locked OOR liquidity

### LINKED ISSUE #607: oor: concurrent send loser reports 'insufficient funds' instead of input-lock contention
## Environment

- `darepo-client`: `cd035b84` (also reproducible on current main `32b33fb5`)
- Network: signet
- Wallet backend: `lnd`
- Ark server: `arkd-signet.testnet.lightningcluster.com:443`

## Summary

When two `darepocli ark send oor` calls race on the same source VTXO, one wins and one fails — but the loser's error message claims **`insufficient funds: need 1000`** with `code=Internal`. The real cause is the custom-input lock (`reserveCustomInputs`) held by the winning call; the wallet had ample balance at request time. Users will chase a non-existent balance bug instead of recognizing transient contention.

## Expected Behavior

- gRPC code should be `Aborted` or `FailedPrecondition` (signalling "retry later"), not `Internal`.
- Message text should name the actual condition — input lock contention — not synthesize a misleading "insufficient funds".
- A loser that retries after the winner releases its lock should succeed.

## Actual Behavior

Setup: signet node b had a single LIVE VTXO of 25,000 sat (plus an unrelated 999-sat dust VTXO). Receiver node a had two fresh receive scripts (different labels, distinct pubkeys, so no duplicate-destination collision):

```
ark oor receive --label "batch-b-rx-1" → pubkey_xonly_hex A
ark oor receive --label "batch-b-rx-2" → pubkey_xonly_hex B
```

Two `ark send oor --pubkey ... --amount 1000` calls fired concurrently from node b targeting A and B respectively.

**Winner:**
```jsonc
{
  "status": "submitted",
  "session_id": "2d905ba8...",
  ...
}
// runs to COMPLETED; consumes the 25,000-sat input,
// produces ~23,000 sat change on node b + 1000 sat on node a.
```

**Loser:**
```jsonc
{
  "error": {
    "code": "Internal",
    "message": "insufficient funds: need 1000"
  }
}
```

The loser's message is misleading. Wallet balance at the time of both requests was 25,999 sat (≥ 1000 by a wide margin). The actual reason is that `reserveCustomInputs` (per `darepod/CLAUDE.md`'s "serializes concurrent calls on the same outpoints" invariant) held the input lock during the winner's MuSig2 dance.

## Reproduction

```sh
# Sender (node B): hold a single LIVE VTXO ≥ amount + dust_limit + op_fee.
# Receiver (node A): get two distinct receive scripts.

A_RX1=$(NODE=a darepocli ark oor receive --label rx-1 | jq -r .pubkey_xonly_hex)
A_RX2=$(NODE=a darepocli ark oor receive --label rx-2 | jq -r .pubkey_xonly_hex)

# fire concurrently
( NODE=b darepocli ark send oor --pubkey $A_RX1 --amount 1000 ) &
( NODE=b darepocli ark send oor --pubkey $A_RX2 --amount 1000 ) &
wait

# Inspect: one COMPLETED, one "insufficient funds: need 1000" with code=Internal.
```

## Notes

Found during a swapdk testing sprint Batch B concurrency pass. The serialization itself appears to be working correctly (one winner, one loser, no double-spend). Only the loser's error wording + gRPC status code are the problem.

### PR DESCRIPTION
**Pull request by bhandras**
## Summary

Fixes #607.

Concurrent OOR sends that race on the same spendable liquidity now distinguish the two important failure modes:

- true spendable-funds shortfall returns a typed insufficient-funds admission error
- liquidity present but locked by an in-flight VTXO operation returns a typed locked-liquidity admission error

The daemon maps those typed errors to caller-actionable gRPC codes instead of collapsing all wallet-selection failures into `Internal`:

- locked liquidity -> `Aborted`, so callers know retrying later is reasonable
- true insufficient spendable funds -> `ResourceExhausted`
- unexpected selection failures still -> `Internal`

This also keeps stale-store/actor-state races covered: if selection sees an apparently live VTXO but the actor rejects the reserve because it is already pending/spending, that path is treated as locked-liquidity contention too.

## Testing

- `go test ./vtxo -run 'TestSelectAndReserveSpend(InsufficientFunds|DoubleExclusion|ReserveRejectedWhenPendingForfeit)|TestRollbackOnPartialSpendFailure|TestSelectAndReserveForfeitInsufficientFunds' -count=1`\n- `go test ./darepod -run TestVTXOAdmissionCode -count=1`\n- `go test ./vtxo ./darepod ./wallet -count=1`\n- `make fmt-changed`\n- `make lint-changed-local`\n- `make fmt-changed-check`\n- `make commitmsg-lint range=origin/main..HEAD`

### CODE REVIEW

**Comment by bhandras** [comment]
Addressed the review tradeoff by documenting the intentional blanket wrapping at the post-selection reserve failure site.

The comment now explains that the common failure here is a stale store view where another actor turn already moved the selected VTXO out of `LiveState`, so we classify it as locked liquidity for retryable caller behavior. It also calls out that unexpected actor/infrastructure errors remain preserved inside the wrapped error for logs.

Revalidated after the amend:

- `go test ./vtxo ./darepod ./wallet -count=1`
- `make fmt-changed`
- `make fmt-changed-check`
- `make commitmsg-lint range=origin/main..HEAD`

Pushed updated head: `532b6058`.

**Comment by bhandras** [comment]
Claude: Re-reviewed updated head `532b6058`. LGTM — good to merge on green CI.

The fold is exactly the option (a) follow-up: a 6-line comment at the post-selection reserve-failure wrap, no logic change (`git diff` is `vtxo/manager.go | 6 ++++++`). The rebase onto current main introduced no drift in the touched files.

## Verified
- **Comment is accurate.** It states the underlying actor/infra error "remains wrapped inside the returned error for logs" — and the wrap does exactly that: `fmt.Errorf("%w: reserve %s %s: %w", ErrVTXOLiquidityLocked, p.label, vtxo.Outpoint, err)` double-wraps, so `errors.Is(..., ErrVTXOLiquidityLocked)` drives the `Aborted` classification while the original `err` stays in the chain for the `WarnS` log emitted just above. The precision loss is confined to the gRPC *code*; the real cause is still observable in logs.
- **Classification logic (unchanged, verified previously).** `ListLiveVTXOs` returns non-terminal VTXOs (Live + PendingForfeit + Forfeiting + Spending), so skipping `VTXOStatusLive` correctly isolates locked liquidity — the `ErrVTXOLiquidityLocked` branch is live in production, not just in the mocked tests. `liveTotal`/`lockedTotal` arithmetic is sound (no double count), and a `ListLiveVTXOs` failure falls through to `Internal`.
- **gRPC mapping** (`Aborted` / `ResourceExhausted` / `Internal`) is applied uniformly to `SendVTXO` and `SendOOR`, and is semantically apt (`Aborted` = retry reasonable; `ResourceExhausted` = genuine shortfall).
- **Build + tests green** locally (`./vtxo ./darepod`).

The only open item from my prior review is resolved. No further concerns.

File: darepod/rpc_server.go [hunk 1]
```diff
@@ -132,6 +132,21 @@ func (r *RPCServer) SignMailboxAuth(ctx context.Context,
...
+	case errors.Is(err, vtxo.ErrVTXOLiquidityLocked):
+		return codes.Aborted
+
+	case errors.Is(err, vtxo.ErrInsufficientSpendableFunds):
+		return codes.ResourceExhausted
+
+	default:
+		return codes.Internal
+	}
+}
```
[the remark this answers is not in the document]
Comment (bhandras) [reply darepod/rpc_server.go:154]: Fixed. `vtxoAdmissionCode` now checks wrapped `context.Canceled` and `context.DeadlineExceeded` before the VTXO sentinel errors, so caller/request cancellation maps to `codes.Canceled` and deadline expiry maps to `codes.DeadlineExceeded` instead of `Internal`.

File: darepod/rpc_server_test.go [hunk 2]
```diff
@@ -41,6 +42,33 @@ func newTestRPCServer() *RPCServer {
...
+				vtxo.ErrInsufficientSpendableFunds),
+		),
+	)
+	require.Equal(
+		t, codes.Internal,
+		vtxoAdmissionCode(
+			errors.New("actor system unavailable"),
+		),
+	)
+}
```
[the remark this answers is not in the document]
Comment (bhandras) [reply darepod/rpc_server_test.go:83]: Fixed. `TestVTXOAdmissionCode` now covers wrapped `context.Canceled` and `context.DeadlineExceeded` and asserts the expected gRPC codes.

File: vtxo/manager.go [hunk 3]
```diff
@@ -685,6 +687,38 @@ func (m *Manager) selectAndReserveVTXOs(ctx context.Context, p reserveParams) (
...
+	}
+
+	var lockedTotal btcutil.Amount
+	for _, desc := range nonTerminal {
+		if desc.Status == VTXOStatusLive {
+			continue
+		}
+
+		lockedTotal += desc.Amount
+	}
```
[the remark this answers is not in the document]
Comment (bhandras) [reply vtxo/manager.go:720]: Fixed. `insufficientLiquidityError` now skips nil descriptors while classifying non-terminal liquidity. I also adjusted the double-exclusion regression test to include a nil row in the non-terminal snapshot so the defensive path is covered.
